scrabble.js

```diff
@@ -1,16 +1,165 @@
 
+const WORDVALUES = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10
+}
+
+const LETTERS = /[A-Z]/
+
 const Scrabble = {
+
   score(word) {
+    const upWord = word.toUpperCase()
+    let score = 0;
+    let wordLetters = this.splitString(upWord);
+
+    if (wordLetters.length > 7) {
+      throw "Character is not a letter!";
+    } else if (wordLetters.length === 0) {
+      throw "Oops! This word is empty!";
+    }
 
+    wordLetters.forEach(function(letter) {
+      if (LETTERS.test(letter)) {
+        score += WORDVALUES[letter];
+      } else {
+        throw "This contains a character is not a letter!";
+      }
+    });
+
+    if (wordLetters.length === 7) {
+      score += 50;
+    }
+    return score;
+  },
+
+  splitString(upWord) {
+    let wordLetters = upWord.split('');
+    return wordLetters;
   },
-  highestScoreFrom(arrayOfWords) {
 
+  highestScoreFrom(arrayOfWords) {
+    if (Array.isArray(arrayOfWords)) {
+      if (arrayOfWords.length === 0) {
+        throw "Oops! No words were given!";
+      } else if (arrayOfWords.length === 1) {
+        return arrayOfWords[0];
+      } else {
+        let topWord = arrayOfWords[0];
+        for (let i = 1; i < arrayOfWords.length; i += 1) {
+          if (this.score(arrayOfWords[i]) > this.score(topWord)) {
+            topWord = arrayOfWords[i];
+          } else if (this.score(arrayOfWords[i]) === this.score(topWord)) {
```

Comment:
You're unnecessarily calculating the scores of each word twice. You should store these in a variable so you don't run these extra processes unnecessarily.